Automattic/wp-calypso
# Domains: Add explanation when trying to transfer domain that is in 60 days transfer lock

### Proposed Changes

Note: This depends on D87294-code.

This PR adds some checks and explanation messages when a customer tries to transfer in a domain that's less than 60 days old, i.e. it has a transfer lock. This should help reduce support interactions caused by domain transfers that can't be done because of transfer locked domains.

See pcYYhz-OU-p2 for more context.

#### Screenshots

- Transfer or connect step
 
![Markup on 2022-09-09 at 20:03:06](https://user-images.githubusercontent.com/5324818/189456787-d16f6dd7-08ed-4756-ad36-b34be35f525d.png)

- Searching for a newly registered domain

![Markup on 2022-09-09 at 20:18:31](https://user-images.githubusercontent.com/5324818/189457821-4bd592b4-3ae2-4bd7-9392-81663daf41ca.png)

- Transfer or connect step during new site creation

![Markup on 2022-09-09 at 20:20:47](https://user-images.githubusercontent.com/5324818/189458047-17906aa5-f0d9-4c32-bad6-8cf444f1ecb4.png)

### Testing Instructions

- Build this branch locally or open the live Calypso link
- Apply D87294-code to your sandbox and sandbox `public-api`
- Go to "Upgrades > Domains"
- Click on "+ Add a domain" and "Use a domain I own"
- Input a domain that was recently registered in another provider
- Ensure the "transfer or connect" step has the transfer option disabled, like in the screenshots above

Please test the domain search form too:

- Go to "Upgrades > Domains" again
- Click on "+ Add a domain" and "Search for a domain"
- Search for a domain that was recently registered in another provider
- Ensure the "&lt;domain&gt; is already registered. Do you own it?" notice appears below the search box

Also please test the domain selection during site creation too:

- Click the "Add new site" button in the left nav bar in Calypso or go to `/start/domains`
- Search for a domain that was recently registered in another provider
- Ensure the "&lt;domain&gt; is already registered. Do you own it?" notice appears below the search box
- If you click the "Do you own it?" link, ensure the "transfer or connect" step has the transfer option disabled, like in the screenshots above

### Pre-merge Checklist
